### PR TITLE
libassert: refactor recipe into v1 and v2 folders

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Specify library name and version:  **libassert/all**

I'm the library author. I'm refactoring the recipe to clean it up a bit by splitting it into v1 and v2 folders as discussed in #23117.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jeremy-rifkin** [comment]
I made two small changes while I was here in addition to the refactor:
- Bumped cpptrace from 0.5.0 to 0.5.1 in the v2 folder
- Updated description and topics metadata in the v2 recipe

**Review by RubenRBS — APPROVED** [review]
Looks great, thanks a lot! :)
The hooks have found one minor issue if you want to fix it, but nothing stopping the PR from getting merged

**Comment by jeremy-rifkin** [comment]
Thanks for pointing that out, I’ll update tonight

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

File: recipes/libassert/v2/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        self.cpp_info.names["cmake_find_package_multi"] = "libassert"
+
+        self.cpp_info.components["assert"].names["cmake_find_package"] = "assert"
+        self.cpp_info.components["assert"].names["cmake_find_package_multi"] = "assert"
+        self.cpp_info.components["assert"].requires = ["cpptrace::cpptrace"]
+        self.cpp_info.components["assert"].libs = ["assert"]
+        if not self.options.shared:
+            self.cpp_info.components["assert"].defines.append("LIBASSERT_STATIC_DEFINE")
+        
+        self.cpp_info.system_libs.append("m")
```
Comment (jwillikers) [inline recipes/libassert/v2/conanfile.py:139]:
```suggestion
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.append("m")
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/libassert/v2/conanfile.py (hunk 1 above)
Comment (jeremy-rifkin) [reply recipes/libassert/v2/conanfile.py]: Thanks!

**Review by jwillikers — APPROVED** [review]

**Pull request merged**